lib/ansible/plugins/connection/ssh.py

```diff
@@ -199,6 +199,12 @@ def wrapped(self, *args, **kwargs):
         remaining_tries = int(C.ANSIBLE_SSH_RETRIES) + 1
         cmd_summary = "%s..." % args[0]
         for attempt in range(remaining_tries):
+            cmd = args[0]
```

Comment:
wouldn't calling `self.reset` be simpler?

Reply:
also .. no need to spawn another ssh client attached to socket for each attempt

Reply:
`self.reset` won't create a new fd/pipe for sshpass.  `self.sshpass_pipe` is only created in `_build_command`, and there is a possibility that the fd changes between creation of the pipe.

I'm not sure what your comment about `also .. no need to spawn another ssh client attached to socket for each attempt` means.  Effectively all this does is open a new pipe that would have been closed, and updates the command to reference the new fd.

Otherwise without this, the fd/pipe is closed, and without updating the command to reference the new fd, we may be referencing the wrong fd

Reply:
nmvd, was confusing the sshpass pipe with ssh one ... but we probably need to regen in reset also.

Reply:
`reset` should not be impacted.  The next time that `_build_command` is used to create a command for a new task it should be ok.  retries is impacted, since we just re-use the already created command from `_build_command`, we don't re-build the command for each retry.